internal/invocation/invocation_service.go

```diff
@@ -78,6 +83,20 @@ func NewService(handler Handler, ed *event.DispatchService, lg logger.LogAdaptor
 			}
 		}()
 	})
+	s.eventDispatcher.Subscribe(EventInvocationStateChanged, stateSubID, func(event event.Event) {
+		e := event.(*InvocationStateChanged)
+		if !e.Enabled {
+			atomic.CompareAndSwapInt32(&s.paused, 0, 1)
```

Comment:
Also, how can we be sure that no invocation is taking place, while we are executing this event callback?

Reply:
The "normal" invocations are not accepted to the invocation channel if the invocation service is paused: https://github.com/yuce/hazelcast-go-client/blob/ad4231ffbac8ec70c57a8c235e732b44b415cb0c/internal/invocation/invocation_service.go#L120